### PR TITLE
Implement basic ui for bank interface

### PR DESCRIPTION
**Pull request by koden-km**
Not fully complete, but covers the basics at this state.
* Customer signup
* Account creation
* Account list
* Deposit
* Withdraw
* Transfer

Refs https://github.com/dogmatiq/example/issues/23

### CODE REVIEW

File: cmd/bank/main.go [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
 	_ "github.com/mattn/go-sqlite3"
 )
 
 func businessDayFromTime(t time.Time) string {
 	return t.Format(messages.BusinessDateFormat)
 }
 
+func readString(reader *bufio.Reader) string {
+	text, _ := reader.ReadString('\n')
+	return strings.Replace(text, "\n", "", -1)
```
[the remark this answers is not in the document]
Comment (koden-km) [reply cmd/bank/main.go]: Yeah just trimming the new line off.

Comment (koden-km) [inline cmd/bank/main.go]: I think your second suggestion is more clear.
```suggestion
	return strings.TrimSuffix(text, "\n")
```

[the remark this answers is not in the document]
Comment (koden-km) [reply cmd/bank/main.go]: Using your second suggestion for this.

**Comment by koden-km** [comment]
Feedback from last weeks UI demo:
- Time advancing quickly
- IDs.. don't show, use business ID (first sword?)
- Validate numbers, including balance check
- Don't show "from" account in "to" list (transfer)
- Ledger
- Breadcrumbs

**Comment by jmalloc** [comment]
> * IDs.. don't show, use business ID (first sword?)

I think just don't produce a UUID, rather produce a random account number like a real bank might. This way we're showing that IDs can be business-relevant, not just database-like meaningless identifiers.

**Comment by koden-km** [comment]
Heh, that should have been `dword` not `sword`, but auto complete changed it.

So you are saying we shouldn't use UUID's at all for the account numbers, or still use UUID's internally but also generate some other account number that is to be displayed and used but customers?

**Comment by jmalloc** [comment]
> So you are saying we shouldn't use UUID's at all for the account numbers

That's right, the fact that the identifiers have meaning to a domain expert can and should be encoded in our domain model.

We can do this in this particular case because the these business identifiers are by-design immutable and non-transferable.

**Comment by koden-km** [comment]
This has been replaced by: https://github.com/dogmatiq/example/pull/57

**Pull request closed without merging**